src/syscall.rs

```diff
@@ -44,3 +44,10 @@ extern "C" fn sys_block(sem: *const Semaphore) -> usize {
 
     0
 }
+
+#[no_mangle]
+extern "C" fn sys_task_spawn(stack_size: usize, entry_point: *const u32, parameter: *mut u32) -> usize {
+    unsafe { task::add_task(stack_size, entry_point, parameter); }
```

Comment:
Instead of having the new task start at the entry point, it might be better to have it start at a helper function, and pass in the entry point and it's parameter as a parameter to the helper function.
I'm imagining the helper function would do two things:
1) Put the raw parameter pointer back into a box, so that when the new task ends, it will go out of scope and deallocate itself.
2) When the task ends, the helper would call the exit system call, to deallocate its stack.

With the way this is written currently, I think we would be just leaking both the parameter and stack data if the entry point function was to return.